src/Common/ZooKeeper/ZooKeeperImpl.cpp

```diff
@@ -596,8 +599,17 @@ void ZooKeeper::sendThread()
                     if (info.request->xid != CLOSE_XID)
                     {
                         CurrentMetrics::add(CurrentMetrics::ZooKeeperRequest);
+
                         std::lock_guard lock(operations_mutex);
-                        operations[info.request->xid] = info;
+                        auto [it, inserted] = operations_map.emplace(
+                            std::piecewise_construct,
+                            std::forward_as_tuple(info.request->xid),
+                            std::forward_as_tuple());
+                        /// Let's actively end the current session if there has been the same xid in the operation queue.
+                        /// But normally the range of xid(1 ... 2~31-1) is large enough not to trigger this exception.
+                        if (!inserted)
+                            throw Exception(Error::ZRUNTIMEINCONSISTENCY, "Have a conflict xid: {}", info.request->xid);
+                        it->second.iter = operations_queue.insert(operations_queue.end(), info);
```

Comment:
> 

In my local test environment, it will be triggered if we increase the newly added setting 'set_zookeeper_xid_value'. Maybe I should add a test case. :)

Reply:
It seems that the code will be executed if we increase the newly added setting 'set_zookeeper_xid_value', doesn't it?